x-pack/qa/oidc-op-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/oidc/C2IdOpTestCase.java

```diff
@@ -56,21 +60,134 @@ public abstract class C2IdOpTestCase extends ESRestTestCase {
 
     // URLs for accessing the C2id OP
     private static final String C2OP_PORT = getEphemeralTcpPortFromProperty("oidc-provider", "8080");
-    private static final String C2ID_LOGIN_API = "http://127.0.0.1:" + C2OP_PORT + "/c2id-login/api/";
-    private static final String C2ID_REGISTRATION_URL = "http://127.0.0.1:" + C2OP_PORT + "/c2id/clients";
-    protected static final String C2ID_AUTH_ENDPOINT = "http://127.0.0.1:" + C2OP_PORT + "/c2id-login";
+    private static final String C2ID_HOST = "http://127.0.0.1:" + C2OP_PORT;
+    protected static final String C2ID_ISSUER = C2ID_HOST + "/c2id";
+    private static final String PROXY_PORT = getEphemeralTcpPortFromProperty("http-proxy", "8888");
+    private static final String C2ID_LOGIN_API = C2ID_HOST + "/c2id-login/api/";
+    private static final String C2ID_REGISTRATION_URL = C2ID_HOST + "/c2id/clients";
+    protected static final String C2ID_AUTH_ENDPOINT = C2ID_HOST + "/c2id-login";
 
-    private static final String ES_PORT = getEphemeralTcpPortFromProperty("elasticsearch-node", "9200");
     // SHA256 of this is defined in x-pack/test/idp-fixture/oidc/override.properties
     private static final String OP_API_BEARER_TOKEN = "811fa888f3e0fdc9e01d4201bfeee46a";
 
     private static Path HTTP_TRUSTED_CERT;
 
+    private static final String CLIENT_SECRET = "b07efb7a1cf6ec9462afe7b6d3ab55c6c7880262aa61ac28dded292aca47c9a2";
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .nodes(1)
+        .setting("xpack.license.self_generated.type", "trial")
```

Comment:
This is all just copied from here: https://github.com/elastic/elasticsearch/pull/96756/files#diff-4efc52cf915c174015f87de02bacc305e91d31fc063fab56d22ec2e0b9a6d674L6